File: packages/core/src/factories/internal/metadata/IMetadataIteratorProps.ts
import ts from "typescript";

import { MetadataCollection } from "../../../schemas/metadata/MetadataCollection";
import { MetadataSchema } from "../../../schemas/metadata/MetadataSchema";
import { MetadataFactory } from "../../MetadataFactory";

export interface IMetadataIteratorProps<Type extends ts.Type = ts.Type> {
  options: MetadataFactory.IOptions;
  checker: ts.TypeChecker;
  components: MetadataCollection;
  errors: MetadataFactory.IError[];
  metadata: MetadataSchema;
  type: Type;
  explore: MetadataFactory.IExplore;
  intersected?: boolean;
}

File: packages/core/src/factories/internal/metadata/IMetadataIteratorProps.go
package metadata

type IMetadataIteratorProps struct {
	Options     MetadataFactoryOptions
	Checker     TypeChecker
	Components  *MetadataCollection
	Errors      *MetadataErrors
	Metadata    *MetadataSchema
	Type        TypeLike
	Explore     MetadataExplore
	Intersected bool
}
